cmd/kubeadm/app/util/config/common.go

```diff
@@ -56,16 +56,16 @@ func validateSupportedVersion(gv schema.GroupVersion, allowDeprecated bool) erro
 	// v1.13: v1alpha3 read-only, writes only v1beta1 config. Errors if the user tries to use v1alpha1 or v1alpha2
 	// v1.14: v1alpha3 convert only, writes only v1beta1 config. Errors if the user tries to use v1alpha1 or v1alpha2
 	// v1.15: v1beta1 read-only, writes only v1beta2 config. Errors if the user tries to use v1alpha1, v1alpha2 or v1alpha3
+	// v1.22: v1beta2 read-only, writes only v1beta3 config. Errors if the user tries to use v1beta1 and older
 	oldKnownAPIVersions := map[string]string{
 		"kubeadm.k8s.io/v1alpha1": "v1.11",
 		"kubeadm.k8s.io/v1alpha2": "v1.12",
 		"kubeadm.k8s.io/v1alpha3": "v1.14",
+		"kubeadm.k8s.io/v1beta1":  "v1.15",
 	}
 
 	// Deprecated API versions are supported by us, but can only be used for migration.
-	deprecatedAPIVersions := map[string]struct{}{
-		"kubeadm.k8s.io/v1beta1": {},
-	}
+	deprecatedAPIVersions := map[string]struct{}{}
```

Comment:
v1beta1 became read-only in 1.15 and became deprecated in 1.17.

v1beta2 became read-only in 1.22, so it will be deprecated at least 1.24+?

Reply:
in this context, read-only means kubeadm no longer writes it as output version, for example when the users does `kubeadm config print...`. internally kubeadm converts **from** this version to it's now preferred v1beta3 (R / W).

also, we are not strictly following this policy:
https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-parts-of-the-api

for a couple of different points:
- we deprecate betas slower
- we enable the latest version as preferred version

i will add a note that v1beta2 will be deprecated in a future release.
after discussions with @fabriziopandini we can do that in 1.22 or later, not strictly decided yet.
users would have ~1 year to transition to v1beta3.